app/assets/javascripts/project_tree_filter.js.coffee

```diff
@@ -0,0 +1,54 @@
+class @ProjectTreeFilter
+  constructor: ->
+
+    # focus text input box
+    $(".tree-filter-input").focus()
+
+    # bind keyup event at text input box
+    $(document).off "keyup", ".tree-filter-input"
+    $(document).on "keyup", ".tree-filter-input", @treeFilter
+
+    $(".tree-filter-input").trigger( "keyup" ) if $(".tree-filter-input").val()
+
+  # filter tree
+  treeFilter: (e) ->
+    searchCharArr = $(this).val().split("")
+    regEx = makeRegEx searchCharArr

```

Comment:
Would it be possible to use a more powerful fuzzy filtering algorithm like https://www.npmjs.com/package/fuzzaldrin-plus ?